Feature/451 bottomsheet UI 개선통합

MrMirror21 opened the pull request:
## Related issue 🛠

[//]: # "해당하는 이슈 번호 달아주기"

- closed #451 
 
## Work Description ✏️

### BottomSheet 컴포넌트 통합 및 개선
- 기존 `BottomSheetLayout` 컴포넌트를 삭제하고 새로운 통합 `BottomSheet` 컴포넌트로 대체
- Compound Component 패턴 적용
  - `BottomSheet.Header`, `BottomSheet.Content`, `BottomSheet.ButtonGroup` 서브 컴포넌트 제공
  - `BottomSheet.Header` : 좌우 아이콘, 헤더 텍스트 정렬 선택 가능
  - `BottomSheet.Content` : `isMaxHeight`으로 최대 길이로 설정 가능(평시 49vh)
  - `BottomSheet.ButtonGroup` : `variant` 통해 버튼 정렬 방식 변경 가능(SINGLE, TWO_HORIZONTAL, TWO_VERTICAL)
- 프로젝트 전반 69개 파일에서 기존 BottomSheet 관련 컴포넌트들을 새로운 통합 컴포넌트로 마이그레이션
- `BottomSheet` 컴포넌트 종합 유닛 테스트 추가(`BottomSheet.test.tsx`)

### 데이터 조회 로직 최적화
- `useResumeData` 훅 개선: 지원자 수락 시 이력서 조회 케이스 분리로 개별 데이터 소스 관리
- 고용주의 이력서 조회 상황별(인재 검색 vs 지원자 확인) 분기 처리 개선

### 기타 개선사항
- 상대 경로 제거 및 파일 구조 정리로 import 일관성 개선
- 마이그레이션 과정 중 레거시 코드에 변경된 디자인 시스템, 리뉴얼된 컴포넌트 적용

## Uncompleted Tasks 😅
N/A

## To Reviewers 📢
- 코드 변경량이 꽤 많으나, 마이그레이션이 대부분이므로, 핵심 변경사항인 `BottomSheet.tsx`, `useResumeData.ts` 파일에 집중해주시면 됩니다.
- `BottomSheet` 통합: 기존 `BottomSheetLayout`을 완전히 대체하는 새로운 `BottomSheet` 컴포넌트는 Compound Component 패턴을 사용하여 `BottomSheet.Header`, `BottomSheet.Content`, `BottomSheet.ButtonGroup`의 서브 컴포넌트를 제공합니다. 제작해야 하는 BottomSheet의 ui에 맞게 조합하여 사용할 수 있습니다.
- 마이그레이션 범위: 이번 작업으로 프로젝트 전반 69개 파일이 수정되었으며, 기존 `BottomSheet` 관련 컴포넌트들이 모두 새로운 시스템으로 통합되었습니다.
- 데이터 조회 최적화: 이전 `useResumeData` 훅 추가 후 고용주의 지원자 이력서 조회 시 7.25가 아닌 7.19 api를 사용해야함을 누락해 이력서를 조회하지 못하는 버그가 있었습니다. 해당 케이스를 커버하도록 dataSourceType에 따라 적절한 API 엔드포인트를 호출하도록 수정, 지원자 수락(applicantAccept), 고용주 인재 검색(owner), 일반 사용자(user) 케이스를 분리했습니다.



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **신규 기능**
  * 하단 시트(BottomSheet) UI 컴포넌트와 Header, Content, ButtonGroup 등 하위 컴포넌트가 추가되었습니다.
  * 회원가입 필수 약관 동의 하단 시트(AgreeBottomSheet) 컴포넌트가 새롭게 제공됩니다.
  * 고용주가 계약별 지원자 이력서를 조회할 수 있는 API 및 관련 React Query 훅이 추가되었습니다.

* **기능 개선 및 리팩터링**
  * 기존 하단 시트 레이아웃(BottomSheetLayout) 기반 컴포넌트들이 새로운 BottomSheet 컴포넌트로 일괄 교체되어 구조가 통일되고, UI 일관성이 향상되었습니다.
  * 여러 버튼, 리스트, 필터, 에러, 로그인 등 다양한 하단 시트들이 새로운 구조로 리팩터링되었습니다.
  * 버튼 스타일 및 타입 지정 방식이 enum 기반으로 변경되어 코드가 간결해졌습니다.
  * 일부 컴포넌트의 props가 변경되거나, 내부 구조가 개선되었습니다.

* **버그 수정**
  * 다양한 하단 시트에서 버튼 스타일 및 동작 일관성이 개선되었습니다.

* **문서화 및 테스트**
  * BottomSheet 및 하위 컴포넌트에 대한 테스트 코드가 추가되어 안정성이 향상되었습니다.

* **기타**
  * 다수의 import 경로가 절대경로로 변경되어 유지보수성이 높아졌습니다.
  * 번역 리소스에 "동의하기" 항목이 추가되었습니다.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

MrMirror21 committed:
✨ feat: BottomSheet 컴포넌트 통합 #451

- Context를 활용해 header, content, buttonGroup으로 나누어 compound components 패턴 사용.
- 기존 BottomSheetLayout의 로직을 BottomSheet 메인 컴포넌트로 통합

MrMirror21 committed:
♻️ refactor: 기존 BottomSheetLayout 파생 컴포넌트 변경 적용 #451

- 기존 BottomSheetLayout 파생 컴포넌트들에 BottomSheet 적용
- 미사용 컴포넌트 제거
- 번역 미지원 요소 번역 추가

MrMirror21 committed:
🔧 chore: 상대 경로 제거, 파일 위치 조정 #451

- 프로젝트 내 상대 경로 제거
- BottomSheet 파일 위치 조정

MrMirror21 committed:
🐛 fix: useResumeData 훅 내 지원자 수락 시 이력서 조회 케이스 분리

- 공고 지원자 수락 시 조회하는 이력서의 경우 일반적인 사용자 id를 통해 조회하는 것이 아닌 해당 고용주-지원자 간 계약 id를 통해 조회하는 특수한 api를 사용하며, 이를 예외 케이스로 분리

MrMirror21 committed:
♻️ refactor: Button 컴포넌트 리뉴얼 적용 #451

MrMirror21 committed:
🔧 chore: 빌드 청크 설정 변경 #451

MrMirror21 committed:
🐛  fix: PostApplyResume 테스트를 useResumeData 훅 변경사항에 맞춰 동기화 #451

- useGetResume, useGetApplicantResume 개별 훅 모킹 제거
- useResumeData 통합 훅 모킹으로 변경
- 기존 3개 테스트 모두 정상 동작 확인
- useCurrentApplicantIdStore 모킹 제거 (더 이상 사용되지 않음)

MrMirror21 committed:
♻️ refactor: 매직넘버 제거 #451

MrMirror21 committed:
✅ test: BottomSheet 유닛 테스트 코드 추가 #451

MrMirror21 committed:
🐛 fix: pr 댓글 내 테스트 커버리지 결과 출력되도록 수정

MrMirror21 committed:
📝 docs: 서비스 데모 영상링크 정상적으로 갱신

naarang reviewed (approved):
LGTM👍👍👍 매직넘버, 상대경로 등의 부분들도 신경써주셨네요!!

SEO-HAN commented:
LGTM 💯💯💯

MrMirror21 committed:
♻️ refactor: 리뷰 피드백 반영 #454

- 구버전 property 제거
- BottomSheet 내 useEffect 루프 현상 방지 처리
- tailwind 동적 클래스 생성 로직 대체
- ResumeDeleteModal 내 useBodyScrollLock 중복 제거
